src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -1067,19 +990,23 @@ const bool ManchesterpatternDecoder::doDecode() {
 				if (pulseIsLong) {
 					// probe clock based preamble
 					if (l == i && i > 0) {
-						int clock = (pdec->pattern[shorthigh] + abs(pdec->pattern[shortlow])) / 2;
-						int pClock = abs(pdec->pattern[pdec->message[l-1]]); 
-						int pClocks = round(pClock / (float)clock); 
+						int pClock = abs(pdec->pattern[pdec->message[l-1]]);
 
-						if (pClocks > 1 && abs(1 - (pClock / (pClocks * (float)clock))) <= 0.02) {
+						// check if level is different
+						if (pClock < maxPulse && ((pdec->pattern[pdec->message[l-1]] > 0) != (pdec->pattern[pdec->message[l]] > 0))) {
```

Comment:
wieso sollte der Wert auf l-1  und l beide größer 0 sein.
Solche Werte sollten nie in den Puffer gelangen. Da wird "eigentlich" ein reset des Puffers aufgerufen, wenn zwei Pegelzeiten mit gleichem pinlevel hintereinander erkannt werden